pkg/dhclient/dhclient.go

```diff
@@ -125,3 +130,106 @@ func WriteDNSSettings(ips []net.IP) error {
 	}
 	return ioutil.WriteFile("/etc/resolv.conf", rc.Bytes(), 0644)
 }
+
+type Lease interface {
+	fmt.Stringer
+	Configure() error
+	Boot() (*url.URL, error)
+	Link() netlink.Link
+}
+
+func lease4(iface netlink.Link, timeout time.Duration, retries int) (Lease, error) {
+	client, err := dhcp4client.New(iface,
+		dhcp4client.WithTimeout(timeout),
+		dhcp4client.WithRetry(retries))
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Attempting to get DHCPv4 lease on %s", iface.Attrs().Name)
+	p, err := client.Request()
+	if err != nil {
+		return nil, err
+	}
+
+	packet := NewPacket4(iface, p)
+	if _, err := packet.Boot(); err != nil {
```

Comment:
Must we always have boot targets? can't we not have a boot target sometimes we dhcp?